packages/docusaurus/src/commands/serve.ts

```diff
@@ -40,17 +45,52 @@ export default async function serve(
     process.exit();
   }
 
+  const {
+    siteConfig: {baseUrl},
+  } = await loadSiteConfig({
+    siteDir,
+    customConfigFilePath: cliOptions.config,
+  });
+
+  const servingUrl = `http://${cliOptions.host}:${cliOptions.port}`;
+
+  /*
+  const proxyServer = httpProxy.createProxyServer({
+    target: `${servingUrl + defaultBaseUrl}`,
+  });
+   */
   const server = http.createServer((req, res) => {
-    serveHandler(req, res, {
+    // Automatically redirect requests to /baseUrl/
+    if (!req.url?.startsWith(baseUrl)) {
```

Comment:
if user access "/" but there is `/baseUrl/`, user could be automatically redirected to that `/baseUrl/`, I find this a convenient way to avoid potential bug reports (even if the log is quite clear...)